### PR TITLE
Basic unit conversion for experiments

### PR DESCRIPTION
**Pull request by FNTwin**
Unit conversion for handling the experiments. Opening the PR while I clean the API and add some tests
It will take care of #9 

## TODO

- [ ] Tests
- [ ] Some cleaning of the API

---

Implement a registry of the most common unit conversions (units.py) and modify the constructor for the datasets to keep track of the current units and provide a way to change them.

### CODE REVIEW

File: src/openqdc/datasets/gdml.py [hunk 1]
```diff
@@ -13,9 +13,11 @@ class GDML(BaseDataset):
     atomic_energies = np.zeros((MAX_ATOMIC_NUMBER,), dtype=np.float32)
 
     __energy_methods__ = [
-        "ccsd",
-        "ccsd(t)",
-        "pbe-ts",
+        "ccsd/cc-pvdz",
+        "ccsd(t)/cc-pvdz",
+        # "pbe+mbd/light", #MD22
```
Comment (prtos) [inline src/openqdc/datasets/gdml.py:18]: We are not using those methods should we?

File: src/openqdc/datasets/geom.py [hunk 2]
```diff
@@ -59,6 +59,10 @@ class GEOM(BaseDataset):
     __name__ = "geom"
     __energy_methods__ = ["gfn2_xtb"]
```
Comment (prtos) [inline src/openqdc/datasets/geom.py:60]: @FNTwin  Do still need a basis set for semi-empirical methods

File: src/openqdc/datasets/molecule3d.py [hunk 3]
```diff
@@ -46,15 +46,19 @@ def _read_sdf(sdf_path, properties_path):
 
 class Molecule3D(BaseDataset):
     __name__ = "molecule3d"
-    __energy_methods__ = ["b3lyp_6-31g*"]
+    __energy_methods__ = ["b3lyp/6-31g*"]
+    # UNITS MOST LIKELY WRONG, MUST CHECK THEM MANUALLY
```
Comment (prtos) [inline src/openqdc/datasets/molecule3d.py:50]: @FNTwin is this already checked?

File: src/openqdc/datasets/gdml.py (hunk 1 above)
Comment (FNTwin) [reply src/openqdc/datasets/gdml.py:18]: It depends if by GDML we are talking about MD17, MD22 or both. Both of them are calculated from AIMD with FHI-aims with All Electron DFT. MD17 uses the pbe functional with vdw-ts, while MD22 uses pbe+mbd with different settings per system. 
I'm going to calculate the isolated atoms energy for  ccsd/cc-pvdz, ccsd(t)/cc-pvdz and I'm creating as we speak the pseudopotential for the pbe functional for the pbe+vdw-ts level of theory on DFT PAW. We would have issue on the many body dispersion as I can't use the correct convergence light/tight so I think just using the pbe+vdw-ts should be good enough

File: src/openqdc/datasets/qm7x.py [hunk 4]
```diff
@@ -39,16 +39,17 @@ class QM7X(BaseDataset):
     # Energy in hartree, all zeros by default
     atomic_energies = np.zeros((MAX_ATOMIC_NUMBER,), dtype=np.float32)
 
-    __energy_methods__ = ["pbe-ts", "mbd"]
+    __energy_methods__ = ["pbe+vdw-ts", "mbd"]
 
     energy_target_names = ["ePBE0", "eMBD"]
 
-    __force_methods__ = ["pbe-ts", "vdw"]
+    __force_methods__ = ["pbe+vdw-ts", "mbd"]
```
Comment (prtos) [inline src/openqdc/datasets/qm7x.py:46]: @FNTwin Are you sure the second force is mbd or just vdw forces?

File: src/openqdc/datasets/waterclusters3_30.py [hunk 5]
```diff
@@ -51,13 +53,13 @@ class WaterClusters(BaseDataset):
 
     # Energy in hartree, all zeros by default
     atomic_energies = np.zeros((MAX_ATOMIC_NUMBER,), dtype=np.float32)
-
+    # need to know where to find the data
```
Comment (prtos) [inline src/openqdc/datasets/waterclusters3_30.py:56]: https://sites.uw.edu/wdbase/database-of-water-clusters/

File: src/openqdc/datasets/geom.py (hunk 2 above)
Comment (FNTwin) [reply src/openqdc/datasets/geom.py:60]: We don't to specify a basis set for semi empirical method as they are fully defined by their name. You don't get to choose the basis set as it is "builded inside" the technique itself.

File: src/openqdc/datasets/molecule3d.py (hunk 3 above)
Comment (FNTwin) [reply src/openqdc/datasets/molecule3d.py:50]: No, I need to download the dataset and run some calculation and manually see what unit they are using as they don't provide any info. This PR should be just to have a quick method to interface openMLIP and convert the datasets between units for experimenting. A more heavy PR with every unit checked and validated will be with the calculated isolated atom energy

File: src/openqdc/datasets/qm7x.py (hunk 4 above)
Comment (FNTwin) [reply src/openqdc/datasets/qm7x.py]: My mistake, corrected. It is a semi empirical tight binding with a many body dispersion correlation and a pbe0 for the dft calculation

File: src/openqdc/datasets/waterclusters3_30.py (hunk 5 above)
Comment (FNTwin) [reply src/openqdc/datasets/waterclusters3_30.py:56]: Ok so you downloaded the entire dataset made from the flexible potential. WIth the isolated atom energy PR I'll have a manual check of the data as with the other unsure dataset

**Pull request merged**